rplugin/python3/deoplete/deoplete.py

```diff
@@ -53,6 +55,13 @@ def __init__(self, vim):
         self.__profile_start = 0
 
     def completion_begin(self, context):
+        if not logger.init:
+            log_file = None
+            if self.__vim.vars['deoplete#enable_debug']:
+                self.debug_enabled = True
+                log_file = self.__vim.vars['deoplete#debug_log']
+            logger.setup(self.__vim, log_file)
```

Comment:
@zchee I was looking closer at things.  How about adding an RPC export for `deoplete_enable_debug`?  This way, the logging can be enabled from VimL where the variable can be checked when `_deoplete()` is called.  `logging.setup()` can be called in the RPC function instead.  With this method, it would be possible to enable logging from a function in VimL without needing to restart.